src/Admin/AdherentAdmin.php

```diff
@@ -409,6 +370,83 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
                     $qb->leftJoin('managed_area.tags', 'managed_area_tag');
                     $qb->andWhere($qb->expr()->in('LOWER(managed_area_tag.name)', $value));
 
+                    return true;
+                },
+            ])
+            ->add('role', CallbackFilter::class, [
+                'label' => 'common.role',
+                'field_type' => ChoiceType::class,
+                'field_options' => [
+                    'choices' => AdherentRoleEnum::toArray(),
+                    'choice_label' => function (string $value) {
+                        return $value;
+                    },
+                    'multiple' => true,
+                ],
+                'show_filter' => true,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return false;
+                    }
+
+                    $where = new Expr\Orx();
+
+                    /** @var QueryBuilder $qb */
+
+                    // Referent
+                    if (\in_array(AdherentRoleEnum::REFERENT, $value['value'], true)) {
+                        $where->add(sprintf('%s.managedArea IS NOT NULL', $alias));
+                    }
+
+                    // Committee supervisor & host
+                    if ($committeeRoles = array_intersect([AdherentRoleEnum::COMMITTEE_SUPERVISOR, AdherentRoleEnum::COMMITTEE_HOST], $value['value'])) {
+                        $qb->leftJoin(sprintf('%s.memberships', $alias), 'ms');
+                        $where->add('ms.privilege IN (:privileges)');
+                        if (\in_array(AdherentRoleEnum::COMMITTEE_SUPERVISOR, $committeeRoles, true)) {
+                            $privileges[] = CommitteeMembership::COMMITTEE_SUPERVISOR;
+                        }
+                        if (\in_array(AdherentRoleEnum::COMMITTEE_HOST, $committeeRoles, true)) {
+                            $privileges[] = CommitteeMembership::COMMITTEE_HOST;
+                        }
+                        $qb->setParameter('privileges', $privileges);
```

Comment:
It looks like `$committeeRoles` can be used directly instead of building `$privileges`.

Perhaps I am missing something.

Reply:
@B-Galati, not really https://github.com/EnMarche/en-marche.fr/blob/c2d028522fe39699b9bca6001064a732fe9a7c18/src/Entity/CommitteeMembership.php#L34

I prefix by `committee_` for what if we need to add `citizen_project_host` for example...

Reply:
Yes I saw what I missed now :+1: 
The constant was not the same, thank you.